Sources/StreamChat/Workers/UserUpdater.swift

```diff
@@ -27,6 +28,70 @@ class UserUpdater: Worker {
             completion?($0.error)
         }
     }
+    
+    /// Blocks the user with the provided `userId`.
+    /// - Parameters:
+    ///   - userId: The user identifier.
+    ///   - completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    ///
+    func blockUser(_ userId: UserId, completion: ((Error?) -> Void)? = nil) {
+        apiClient.request(endpoint: .blockUser(userId)) {
+            switch $0 {
+            case .success:
+                self.database.write({ session in
+                    session.currentUser?.blockedUserIds.insert(userId)
+                    if let channel = ChannelDTO.directMessageChannel(
+                        participantId: userId,
+                        context: self.database.writableContext
+                    ) {
+                        try session.channel(cid: .init(cid: channel.cid))?.isHidden = true
+                    }
```

Comment:
This should be enough: 🤔 
```suggestion
                    if let channel = ChannelDTO.directMessageChannel(
                        participantId: userId,
                        context: self.database.writableContext
                    ) {
                       channel.isHidden = true
                    }
```

Reply:
Nice! Updated 🤠